functions/completed-checks/v1.js

```diff
@@ -0,0 +1,256 @@
+'use strict'
+
+const moment = require('moment')
+const sqlService = require('less-tedious')
+const uuid = require('uuid/v4')
+const winston = require('winston')
+const { TYPES } = require('tedious')
+
+winston.level = 'error'
+const config = require('../config')
+sqlService.initialise(config)
+const azureStorageHelper = require('../lib/azure-storage-helper')
+const sqlUtil = require('../lib/sql-helper')
+const markingService = require('./marking.service')
+
+// SQL server vars
+const checkResultTable = '[checkResult]'
+const schema = '[mtc_admin]'
+const checkStatusTable = '[checkStatus]'
+const checkTable = '[check]'
+
+const v1 = {
+  process: async function process (logger) {
+    await handleCompletedCheck(logger)
+  }
+}
+
+// Table Storage
+const azureTableService = azureStorageHelper.getPromisifiedAzureTableService()
+
+async function handleCompletedCheck (context, completedCheckMessage) {
+  context.log('completed-check: message received', completedCheckMessage.checkCode)
+
+  let checkData
+  try {
+    checkData = await sqlUtil.sqlFindCheckByCheckCode(completedCheckMessage.checkCode)
+  } catch (error) {
+    context.log.error(`completed-check: ERROR: failed to retrieve checkData for checkCode: ${completedCheckMessage.checkCode}`)
+    throw error
+  }
+
+  if (!canCompleteCheck(checkData.code)) {
+    context.log.error(`completed-check: ERROR: check ${completedCheckMessage.checkCode} is not in a correct state to be completed. Current state is ${checkData.code}`)
+    return // consume the message
```

Comment:
Or would it be better to `throw` here and let the message end up on the poison queue?

Reply:
@jon-shipley it does make sense, I see no reason why we wouldn't want that ending up in the poison queue if the check status code is not present or one of the ones expected.